src/verify_cert.rs

```diff
@@ -341,11 +336,32 @@ fn check_basic_constraints(
     }
 }
 
+/// Extended Key Usage (EKU) of a certificate.
+pub enum ExtendedKeyUsage {
+    /// The certificate must contain the specified [`KeyPurposeId`] as EKU.
+    Required(KeyPurposeId),
+
+    /// If the certificate has EKUs, then the specified [`KeyPurposeId`] must be included.
+    RequiredIfPresent(KeyPurposeId),
+}
+
+/// An OID value indicating the Extended Key Usage (EKU) of the certificate.
```

Comment:
Small nit: I think it might be clearer written this way that this is a purpose that might be asserted as part of an EKU extension?

```suggestion
/// An OID value indicating a Extended Key Usage (EKU) key purpose.
```